src/repr/src/explain_new.rs

```diff
@@ -134,6 +143,45 @@ where
     }
 }
 
+/// Creates a type whose [`fmt::Display`] implementation outputs each item in
+/// `iter` separated by `separator`.
+pub fn separated_text<'a, I, C>(separator: &'a str, iter: I) -> impl fmt::Display + 'a
```

Comment:
Can you add a comment clarifying the difference between this and the version in `mz_ore::str`?